src/node_options.cc

```diff
@@ -24,6 +24,44 @@ enum {
 #endif
 };
 
+// Normalize the argument vector by exploding
+// multiple options (if any). For example
+// "node -abc --scm git" -> "node -a -b -c --scm git"
+// implementation was adapted from
+// https://github.com/clibs/commander/blob/master/src/commander.c
+static char ** normalize_args(int *argc, char **argv) {
+  int size = 0;
+  int alloc = *argc + 1;
+  char **nargv = static_cast<char **>(malloc(alloc * sizeof(char *)));
+  int i;
+
+  for (i = 0; argv[i]; ++i) {
+    const char *arg = argv[i];
+    size_t len = strlen(arg);
+
+    // short flag
+    if (len > 2 && '-' == arg[0] && !strchr(arg + 1, '-')) {
+      alloc += len - 2;
+      nargv = static_cast<char **>(realloc(nargv, alloc * sizeof(char *)));
+      for (size_t j = 1; j < len; ++j) {
+        nargv[size] = static_cast<char *>(malloc(3));
+        snprintf(nargv[size], sizeof(nargv[size]), "-%c", arg[j]);
+        size++;
+      }
+      continue;
+    }
```

Comment:
ok, so I think we should go with https://github.com/nodejs/io.js/pull/1862 for now and just revert the commit.